File: core/LongestConsecutiveSequence/LongestConsecutiveSequence.java
package com.dbc;

import java.util.HashMap;
import java.util.Map;

public class LongestConsecutiveSequence {
    public int longestConsecutive(int[] nums) {
        Map<Integer, Boolean> map = new HashMap<>();
        int res = 0;
        for (int num : nums) map.put(num, false);
        for (int num : nums) {
            if (!map.get(num)) {
                int start = num - 1, end = num + 1;
                while (map.containsKey(start)) {
                    map.put(start, true);
                    start--;
                }
                while (map.containsKey(end)) {
                    map.put(end, true);
                    end++;
                }
                map.put(num, true);
                res = Math.max(res, end - start - 1);
            }
        }
        return res;
    }
}

File: core/LongestConsecutiveSequence/LongestConsecutiveSequence.py
from typing import List


def longestConsecutive(nums: List[int]) -> int:
    nums_map, res = {num: False for num in nums}, 0
    for num in nums:
        if nums_map[num]:
            continue
        count, temp = 0, num
        while temp in nums_map:
            count += 1
            nums_map[temp] = True
            temp += 1
        temp = num - 1
        while temp in nums_map:
            count += 1
            nums_map[temp] = True
            temp -= 1
        res = max(res, count)

    return res


